system/Database/BaseBuilder.php

```diff
@@ -1358,9 +1359,9 @@ public function set($key, $value = '', ?bool $escape = null)
             if ($escape) {
                 $bind = $this->setBind($k, $v, $escape);
 
-                $this->QBSet[$this->db->protectIdentifiers($k, false, $escape)] = ":{$bind}:";
+                $this->QBSet[$this->db->protectIdentifiers($k, false)] = ":{$bind}:";
```

Comment:
This change changes the behavior.
Even if `$escape` is false, `protectIdentifiers()` escapes the identifier.
https://github.com/codeigniter4/CodeIgniter4/blob/c05f5a6dae2b08e9ebf82d5f7888dd8fcbff5813/system/Database/BaseConnection.php#L954-L958

Users can't stop escaping.

Reply:
I decided that there was no case where the column name in the set clause did not need to be escaped, so I turned off the unescaping process.

Are there any use cases where the column names in the set clause do not need escaping?

Reply:
Good point. But it seems it has nothing to do with this Oracle driver.
Can you separate the changes that are not specific to the Oracle driver as another PR?
This PR is too big and takes too much time to review.

Reply:
Sorry to make your review so difficult.
I will split it up.💪

Reply:
This change was merged into CI, so I rebased it.